src/libstd/f64.rs

```diff
@@ -970,6 +970,32 @@ impl f64 {
         0.5 * ((2.0 * self) / (1.0 - self)).ln_1p()
     }
 
+    /// Returns max if self is greater than max, and min if self is less than min.
+    /// Otherwise this returns self.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(clamp)]
+    /// use std::f64::NAN;
+    /// assert!((-3.0f64).clamp(-2.0f64, 1.0f64) == -2.0f64);
+    /// assert!((0.0f64).clamp(-2.0f64, 1.0f64) == 0.0f64);
+    /// assert!((2.0f64).clamp(-2.0f64, 1.0f64) == 1.0f64);
+    /// assert!((NAN).clamp(-2.0f64, 1.0f64).is_nan());
+    /// ```
+    ///
+    /// # Panics
+    /// Panics if min > max, min is NaN, or max is NaN.
+    #[unstable(feature = "clamp", issue = "44095")]
+    #[inline]
+    pub fn clamp(self, min: f64, max: f64) -> f64 {
```

Comment:
I'm a Rust beginner, but I think, this should also work. I don't like the ```mut```.

```rust
if x < min { min }
if x > max { max }
else { self }
```

Reply:
That was the original implementation.  This specific code pattern was chosen because it yields a highly efficient assembly implementation.  This was actually the source of quite a bit of conversation before we settled on this.

Reply:
Oh, interesting and ```let x```?

Reply:
Yeah, and that.

Reply:
Holy caroly (this is actually no real word, but it rhymes)!